lib/wallaby/phantom/driver.ex

```diff
@@ -333,6 +333,135 @@ defmodule Wallaby.Phantom.Driver do
+  """
+  def accept_alert(%Session{}=session, fun) do
+    script = """
+    var page = this;
+    page.__onAlertDefault = page.onAlert;
+    page.onAlert = function(msg) {
+      page.__alertMessage = msg;
+      page.onAlert = page.__onAlertDefault;
+    }
+    return "ok";
+    """
+    {:ok, "ok"} = execute_phantom_script(session, script)
+    fun.(session)
+    script = """
+    var page = this;
+    page.onAlert = page.__onAlertDefault;
+    return ["ok", page.__alertMessage];
+    """
+    {:ok, ["ok", message]} = execute_phantom_script(session, script)
+    message
+  end
+
+  @doc """
+  Accept one confirm triggered within `fun` and return the confirm message.
+  """
+  def accept_confirm(%Session{}=session, fun) do
+    handle_confirm(session, fun, true)
+  end
+
+  @doc """
+  Dismiss one confirm triggered within `fun` and return the confirm message.
+  """
+  def dismiss_confirm(%Session{}=session, fun) do
+    handle_confirm(session, fun, false)
+  end
+
+  defp handle_confirm(%Session{}=session, fun, return_value) do
+    script = """
+    var page = this, returnVal = arguments[0];
+    page.__onConfirmDefault = page.onConfirm;
+    page.onConfirm = function(msg) {
+      page.__confirmMessage = msg;
+      page.onConfirm = page.__onConfirmDefault;
+      return returnVal;
+    }
+    return "ok";
+    """
+    {:ok, "ok"} = execute_phantom_script(session, script, [return_value])
+    fun.(session)
+    script = """
+    var page = this;
+    page.onConfirm = page.__onConfirmDefault;
+    return ["ok", page.__confirmMessage];
+    """
+    {:ok, ["ok", message]} = execute_phantom_script(session, script)
+    message
+  end
+
+  @doc """
+  Accept one prompt triggered within `fun` with the specified `input_value`
+  and return the confirm message.
+  """
+  def accept_prompt(%Session{}=session, input_value, fun) do
+    handle_prompt(session, fun, input_value, true)
+  end
+
+  @doc """
+  Dismiss one prompt triggered within `fun` and return the confirm message.
+  """
+  def dismiss_prompt(%Session{}=session, fun) do
+    handle_prompt(session, fun, nil, false)
+  end
+
+  defp handle_prompt(%Session{}=session, fun, return_value, use_default) do
+    script = """
+    var page = this, returnVal = arguments[0], useDefault = arguments[1];
+    page.__onPromptDefault = page.onPrompt;
+    page.onPrompt = function(msg, defaultVal) {
+      page.__promptMessage = msg;
+      page.onPrompt = page.__onPromptDefault;
+      if (useDefault) {
+        return returnVal || defaultVal;
+      } else {
+        return returnVal;
+      }
+    }
+    return "ok";
+    """
+    {:ok, "ok"} = execute_phantom_script(session, script, [return_value, use_default])
+    fun.(session)
+    script = """
+    var page = this;
+    page.onPrompt = page.__onPromptDefault;
+    return ["ok", page.__promptMessage];
+    """
+    {:ok, ["ok", message]} = execute_phantom_script(session, script)
```

Comment:
How can I best access the output JS Logger? If we use the "normal" `execute_script` function to trigger an error, it will not crash. But it returns something like this, which at least includes the original error message.


```
IO.inspect execute_script(session, "boom")

%{"class" => "org.openqa.selenium.WebDriverException",
  "message" => "{\"errorMessage\":\"Can't find variable: boom\",\"request\":{\"headers\":{\"Content-Length\":\"27\",\"Content-Type\":\"text/json, text/json\",\"Host\":\"localhost:51992\",\"User-Agent\":\"hackney/1.6.6\"},\"httpVersion\":\"1.1\",\"method\":\"POST\",\"post\":\"{\\\"script\\\":\\\"boom\\\",\\\"args\\\":[]}\",\"url\":\"/execute\",\"urlParsed\":{\"anchor\":\"\",\"query\":\"\",\"file\":\"execute\",\"directory\":\"/\",\"path\":\"/execute\",\"relative\":\"/execute\",\"port\":\"\",\"host\":\"\",\"password\":\"\",\"user\":\"\",\"userInfo\":\"\",\"authority\":\"\",\"protocol\":\"\",\"source\":\"/execute\",\"queryKey\":{},\"chunks\":[\"execute\"]},\"urlOriginal\":\"/session/5cb946a0-19c4-11e7-aef6-3973bb41a47b/execute\"}}",
  "screen" => "<Large Base64 encoded string ...>"}
```


Since both `execute_script` and `execute_phantom_script` internally use `check_logs!` I suppose that is what you mean with the internal JS Logger. AFAICS both these return values already come from the logger.

It would be really nice to raise an error in Wallaby if an exception is raised in  `execute_script` or `execute_phantom_script`.